### ISSUE TITLE
Suggestions - Drag and Drop text, line highlights, and drop target split view panes

### ISSUE THREAD

**Issue by OffLuffy** [adobe/brackets#10325]
I'm new to this program, so sorry if there's a way to do something already that I've suggested!

---
##### Drag & Drop Text:
The ability to drag and drop selected text to move it around easier

---
##### Active Line Highlighting:
The current line (lines if more than one row is selected) have a different background color. If another pane in split view is active, the active line on the other panes should remain highlighted to act as somewhat of a place marker if you're referring to the other pane frequently without making it the active pane.

---
##### Drop Target for Split View Panes
The ability to drag and drop files (or file names in the open files pane into panes) to open it in that pane

---
Great work so far! Winning my favor over N++ even without above suggestions, HTML files or not.

**Comment by OffLuffy** [comment]
An a latent note, another suggestion (which I never see employed in any text editor, so maybe a reason that it isn't) is to allow the use of `Shift` + `Arrow keys` to modify the bounds of block-selections (`alt` + dragging to select text) instead of having it revert to a normal selection mode. This would feel more natural after making a block-selection.

**Comment by peterflynn** [comment]
**Drag/drop text:** Please upvote [Drag and Drop to move selected text within editor](https://trello.com/c/io24Byhm/910-drag-and-drop-to-move-selected-text-within-editor) on our feature backlog.

**Active Line Highlight for inactive pane/window:** Makes sense. It should be possible to do this in a theme, but it seems reasonable to do it in core too.  @larz0, thoughts?  We'd probably want a dimmer active line highlight when inactive (that's what other editors that don't hide the highlight when inactive do).

**Drop target for split panes:** It should work this way already when dragging in a file from the OS folder view. Please let us know if not.  For files that are already open, you don't have to drag them all the way to the other pane -- you can just drag them between sections in the open files list in the upper-left.

**Keys to modify block selection:** Sounds like a good candidate for someone to write an extension -- so adding the 'extension idea' tag for that.

Note: In the future, it would be good to file unrelated things as separate bugs.  But also please check the backlog to see if it's already tracked there.

**Comment by redmunds** [comment]
Drag/drop text is [actively being worked on](https://github.com/adobe/brackets/pull/9584) (and is close to completion), so no need to vote!

**Comment by aringot** [comment]
Anyone working on the active line highlight ? I would like to do it

**Comment by larz0** [comment]
@peterflynn we're using 0.7 opacity to dim the entire inactive pane so using the same active line color could be okay.

**Comment by peterflynn** [comment]
@larz0 We're not dimming the entire pane anymore (as of 1.2), so we'll need to specifically mute the active line highlight.

@aringot As far as I know no one's working on that yet -- feel free!

**Comment by aringot** [comment]
@peterflynn Ok, great !
Any hint on where I should start ? (I found references to the active line in the LESS files, but I can't find where the active line is defined)

**Comment by aringot** [comment]
Do we really need to put a dimmer highlight ? I mean, as long as nothing is dimmed anymore in the inactive pane, there is no reason to have the active line dimmed. Maybe I should only make the inactive pane keep the highlight on the line that was active ?
(even if I would personally prefer the entire inactive pane to be a bit dimmer)

**Comment by peterflynn** [comment]
@aringot Yes, we want it to be dimmer so it's clear which pane has focus.  All the other cursor/selection-related bits of UI change appearance when focus is lost -- cursor is hidden, selection changes color, etc.

**Comment by peterflynn** [comment]
If you want the entire pane to be dimmer, you can install the Zen Panes extension. But per #9785 this won't be the default behavior in Brackets anymore.

**Comment by aringot** [comment]
@peterflynn Thanks for the link !
And indeed, my bad, selection does change color, html tags don't.

I understood how to get the highlight in the inactive pane, now I'm trying to find good colors... Which is kind of hard as the highlight is already really dim ! (And I'll try to code the LESS rules more nicely)

![dimmerhighlightdefaulttheme](https://cloud.githubusercontent.com/assets/5434864/6163668/32babaca-b24a-11e4-8dbe-7559268c33ad.JPG)
